File: Programmers/Python/Code/제일 작은 수 제거하기.py
# 풀이1
def solution(arr):
    answer = []
    if len(arr) == 1:
        answer = [-1]
    else:
        min = arr[0]
        index = 0
        for idx, value in enumerate(arr):
            if value < min:
                min = value
                index = idx
        answer = arr
        answer.pop(index)
                
    return answer

# 풀이2
def solution2(arr):
    m = min(arr)
    arr.remove(m)
    if arr == []:
        answer = [-1]
    else:
        answer = arr
    return answer

File: Programmers/JavaScript/제일 작은 수 제거하기.js
function solution(arr) {
    if (arr.length === 1) {
        return [-1]
    }

    var minNum = arr[0];
    var idx = 0;

    for (var i = 1; i < arr.length; i++) {
        if (minNum > arr[i]){
            minNum = arr[i];
            idx = i;
        }
    }
    arr.splice(idx, 1);
    
    return arr;
}

// 풀이2
function solution2(arr) {
    if (arr.length === 1 ) {
        return [-1]
    }
    const minValue = Math.min.apply(null, arr)
    const index = arr.findIndex(value => value === minValue)
    arr.splice(index, 1)

    return arr;
}